src/posthog-core.ts

```diff
@@ -408,6 +414,8 @@ export class PostHog {
             this.config.persistence === 'sessionStorage'
                 ? this.persistence
                 : new PostHogPersistence({ ...this.config, persistence: 'sessionStorage' })
+
+        // should I store the initial person profiles config in persistence?
```

Comment:
question

Reply:
The way I've handled this so far is to save whether or not they've sent an event with person processing. If they have then we persist `ENABLE_PERSON_PROCESSING` to true.

I'm just wondering whether saving their config gives us anything beyond this.

Here's a couple of cases:

Case 1
* posthog-js starts up with the user setting 'always'
* some events are sent, with person profiles, so `ENABLE_PERSON_PROCESSING` is set
* user closes the browser
* new visit, posthog-js starts up with no setting so uses the default `identified_only`
* some events are sent, as `ENABLE_PERSON_PROCESSING` was set then these have person profiles

Case 2
* posthog-js starts up with the user setting 'always'
* no events are sent, so `ENABLE_PERSON_PROCESSING` is not set
* user closes the browser
* new visit, posthog-js starts up with no setting so uses the default `identified_only`
* some events are sent, these are anonymous (if we'd persisted the setting from before, they'd have person profiles)

In case 2 there's a difference, but I'd argue that those events should be sent as anonymous. Wdyt?

Reply:
TLDR no I don't think so :D

Reply:
Ooooook so it turns out we don't actually handle this case correctly. I added a fix and some tests to this PR. I hope you don't mind me piggybacking like this, I can very much move it to its own PR if you prefer.

Reply:
No this is great, thanks!